src/main/java/com/google/netpcapanalysis/dao/PCAPParserDaoImpl.java

```diff
@@ -0,0 +1,173 @@
+package com.google.netpcapanalysis.dao;
+
+import com.google.netpcapanalysis.models.PCAPdata;
+import java.util.*; 
+import java.io.*;
+import java.lang.*;
+
+import io.pkts.PacketHandler;
+import io.pkts.Pcap;
+import io.pkts.buffer.Buffer;
+import io.pkts.packet.Packet;
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.packet.IPPacket;
+import io.pkts.protocol.Protocol;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.netpcapanalysis.interfaces.dao.PCAPDao;
+import com.google.netpcapanalysis.interfaces.dao.PCAPParserDao;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/*
+* Given a filename and local IP address, parses out individual packets as PCAPdata objects and puts in Datastore.
+* We define myip to be the IP address of the local machine that was used to create the PCAP file, 
+     and outip to be all of the IP addresses that machine connected with.
+   * allPCAP HashMap indexes all unique outips in a PCAP file to a PCAPdata object that stores needed attributes, 
+     including the number of connections between myip and outip 
+   * We load datastore with all of the unique PCAPdata objects from our HashMap with a given entity tag.
+   * Hard-coded the source IP address (myip) -- need to find a way to retrieve myip (look at first packet, ask user, use Whatsmyip?)
```

Comment:
This is something we should ask Philip on the chat room. Generally trying to extract user's IP from the PCAP file sounds intrusive or even asking them.
What about the public pcap files. Is it possible to use a simple heuristic like check the first few rows and determine which is the common ip and set that as the myip. Or you can do something fancy like learning on the fly, i.e. as you read more rows you can correct the assumption.
Also if you only have 2 IPs in your capture it will be impossible to tell i guess. So you could just store them as directional being agnostic of MYIP. Then yo can always compute the undirectional frequency from directional data.